Automattic/wp-calypso
# Signup: Reset Signup Redux state on complete.

Signup's Redux state wasn't cleared out when you finished signup. This caused step data to be repopulated when the user returned again after some time.

This PR adds a reset mechanism for the whole signup state through the `SIGNUP_COMPLETE_RESET` reducer. It is currently fired at the `SignupDependencyStore` `reset()` event. It is called exactly when Signup needs to be reset. 

To test:

1. Checkout branch or use the Calypso.live version.
2. Start signup.
3. Try to get the Site Title step (AB test).
4. Submit Site Title step.
5. Check the step added it's data to Indexed DB in `signup.steps.siteTitle` 
6. Go through the end of signup and reach the Processing screen after filling the username form.
7. Do NOT click Continue.
8. Check again the Indexed DB if `signup.steps.siteTitle` has been cleared properly (make sure to "reload" the screen with the little reload button under the Indexed DB table.

cc @michaeldcain @coreh
